### DISCUSSION TITLE
New GitHub Action not displayed on the Actions tab

### DISCUSSION THREAD

**Discussion by GuillaumeCleme** [community/community#25179] [Actions]
<p>The repository is the following: <a href="https://github.com/GuillaumeCleme/AEM-Remote-ClientLib-Proxy" rel="nofollow noopener">https://github.com/GuillaumeCleme/AEM-Remote-ClientLib-Proxy</a></p>
<p>Action Workflows taken from available actions-starter files and created using the Actions tab “New Workflow” option. I tried triggering multiple changes, fixing line endings, tabbing, etc. but the <code>mvndeploy.yml</code> action never showed up under the Actions tab.</p>
<p>Action files are located at: <a href="https://github.com/GuillaumeCleme/AEM-Remote-ClientLib-Proxy/tree/develop/.github/workflows" rel="nofollow noopener">https://github.com/GuillaumeCleme/AEM-Remote-ClientLib-Proxy/tree/develop/.github/workflows</a></p>
<p>Any help is appreciated.</p>
<p>Cheers,</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/guillaumecleme">@guillaumecleme</a> ,</p>
<p>I checked your repository and the releases you have created, and I found few questions:</p>
<p><strong>1)</strong> The workflow YAML file " <strong>mvndeploy.yml</strong>" is only located on the <strong>develop</strong> branch.</p>
<p><strong>2)</strong> All the releases you have created are from the  <strong>master</strong>  branch.</p>
<p>Here are few points need you pay attention to:</p>
<p><strong>1)</strong> After you have created a new workflow, this workflow will not be listed on " <strong>Actions</strong>" tab until it is triggered for the first time.</p>
<p><strong>2)</strong> The <strong>GITHUB_REF</strong> of the release event is a tag ref, and Github will only reference the workflow YAML files on this tag ref, and the tag ref is created from the specified branch (master in your cases).</p>
<p><strong>3)</strong> You can’t use the event occurs on a branch/tag ref to trigger the workflows located on another branch/tag ref.</p>
<p>So, my suggestions:</p>
<p><strong>1)</strong> If you want to trigger the release workflow on the  <strong>develop</strong> branch, when you create a new release, you should select the tag created from the  <strong>develop </strong> branch.</p>
<p><img src="https://user-images.githubusercontent.com/46307996/181089262-0bfb8836-3aab-4fda-a148-7cee77cb3ba4.png" alt="example.PNG" data-base62-sha1="wG5GlV4oFkXsGdJcNu8GM2Qldx4" width="453" height="164"></p>
<p><strong>2)</strong> Similarly, if you create release from the  <strong>master</strong>  branch, and want to trigger the release workflow, you should make sure the YAML file also has been existing on the  <strong>master</strong>  branch, and select the tag created from the <strong>master</strong>  branch when creating a new release.</p>
<p>More details about Release event, you can reference here: <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#release-event-release" rel="nofollow noopener">https://help.github.com/en/actions/reference/events-that-trigger-workflows#release-event-release</a></p>

**Reply by stephen-oconnor-cp** [reply]
I'm having a similar problem.  I pushed to my one and only branch, and in the .github/workflows folder I have one .yml file.

I'm wondering why it's not showing up so to even trigger manually.   You mention it doesn't show up until you run it once, so where do I run it once?  I'm confused.  

Is github doing some validation and only shows the workflows that have passed some validation and perhaps I have a syntax error in my file?

**Reply by Staubgeborener** [reply]
@stephen-oconnor-cp I found a dirty workaround with something like that:
```
  push:
    branches:
      - test
 ```
You have to simply push nonsense into the `test` branch of your repository in order to trigger the workflow. After that you will see the workflow in the actions tab.

The `workflow_dispatch` event only works in [default branch](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow#configuring-a-workflow-to-run-manually). Same for `schedule`.

**Reply by rdhmdhl** [reply]
same problem here. tried creating a test branch and updating the yml file for that test branch. no luck.

**Reply by Everkent** [reply]
I have the same issue, i am not trying to trigger the workflow on the default branch being Master. i have a develop branch where i need to test the workflow before i can make PR to the master branch.  I have also tried to trigger on pull_request to the develop branch but still no workflow showing on the Actions tab. added workflow dispatch as well no luck:

name: Test Convert YML to CSV for Develop-Ben Branch

on:
  push:
    branches:
      - develop-ben
    paths:
      - 'licsvr-cac/vars/services_new.yml'
  workflow_dispatch:

**Comment by GuillaumeCleme** [comment]
<p>Hi <a class="mention" href="/brightran">@brightran</a>,</p>
<p>I tried a few of your suggestions, but still the Action did not trigger. I had previously merged the <code>mavenpublish.yml</code> file to /master when creating the 1.0.0 release. See this <a href="https://github.com/GuillaumeCleme/AEM-Remote-ClientLib-Proxy/commit/f3604118e434faa5097f6ce642ffbe63f5529f28" rel="nofollow noopener">commit</a>.</p>
<p>After a lot of testing, triggering changes, etc. I discovered that the real issue may have been the release process. During the release process, I followed these steps:</p>
<ol>
<li>Create a release branch</li>
<li>Prepare the project</li>
<li>Merge the release branch to /master</li>
<li>Tag /master</li>
<li>Push to GitHub remote</li>
<li>GitHub then interpreted the tag as a release, but did not trigger the “release” process and then never called the Action</li>
</ol>
<p>As for your comment regarding the Action not showing up until it was triggered for the first time, I was able to confirm this by creating a release manually through the GitHub user interface. </p>
<p>Thank you for the link on workflow events. I will look at refining the event trigger to hopefully run when a tag is pushed.</p>

**Comment by GuillaumeCleme** [comment]
<p>TL;DR</p>
<p>Altough <strong>tags</strong> show up as <strong>releases</strong> in the Github UI, creating and pushing a <strong>tag</strong> does not trigger a  <strong>release</strong> creation event! Since the action does not trigger the event that my GitHub Action was listening to, it never ran and therefore never showed up in the Actions section.</p>
<p>By configuring the action workflow to instead listen to <strong>tag</strong> push events, I was able to achieve the desired result.</p>
<p><a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes</a></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/guillaumecleme">@guillaumecleme</a> ,</p>
<p>Well, according to your reply, I think you have some misunderstandings about Release event, or you confuse Release event with Push event.</p>
<p>When you push a new tag to the remote, this activity is a type of Push event, “push” does not only mean push commits, it also includes push tags.</p>
<p>The Release event will be triggered from the time when you create a new release with a specified tag (an existing tag or create a new tag).</p>

**Comment by GuillaumeCleme** [comment]
<p><a class="mention" href="/brightran">@brightran</a> correct. What was confusing is that pushing a tag also shows up as a release in the UI.</p>

**Reply by whatisor** [reply]
I guess they should open source github then we can fix such a simple bug.
